contents/select.md

```diff
@@ -0,0 +1,191 @@
+```
+
+To select everything _except_ one or more columns, use `Not` with either a single column name:
+
+```jl
+s = """select(responses(), Not(:q5))"""
+sco(s, process=without_caption_label)
+```
+
+Or, with multiple column names:
+
+```jl
+s = """select(responses(), Not([:q4, :q5]))"""
+sco(s, process=without_caption_label)
+```
+
+It's also fine to combine column names with `Not`:
+
+```jl
+s = """select(responses(), :q5, Not(:id))"""
+sco(s, process=without_caption_label)
+```
+
+Note how `q5` is now the first column.
+But, there is a more clever way to do this, and that is with `:`.
+The colon `:` can be thought of as "all the columns that we didn't include yet".
+For example:
+
+```jl
+s = """select(responses(), :q5, :)"""
+sco(s, process=without_caption_label)
+```
+
+Or, to put `q5` at the second position[^sudete]:
+
+[^sudete]: Thanks to Sudete on [Discourse](https://discourse.julialang.org/t/pull-dataframes-columns-to-the-front/60327/4) for this suggestion.
+
+```jl
+s = "select(responses(), 1, :q5, :)"
+sco(s, process=without_caption_label)
+```
+
+Even renaming columns is possible via `select` via `source => target`:
+
+```jl
+s = """select(responses(), 1 => "participant", :q1 => "age", :q2 => "nationality")"""
+sco(s, process=without_caption_label)
+```
+
+which, thanks to the `...` operator, we can also write as:
+
+```jl
+s = """
+    renames = (1 => "participant", :q1 => "age", :q2 => "nationality")
+    select(responses(), renames...)
+    """
+sco(s, process=without_caption_label)
+```
+
+> **Note:** Another example of splatting via the `...` operator is:
+> ```jl
+  s = """
+      V = ["a", "b", "c"]
+      joinpath(V...)
+      """
+  scob(s)
+  ```
+
+## Types and Missing Data {#sec:missing_data}
+
+```{=comment}
+Try to combine with transformations
+
+categorical
+allowmissing
+disallowmissing
+```
+
+As discussed in @sec:load_save, `CSV.jl` will do its best to guess data types for your data.
+However, this won't always work perfectly.
+In this section, we show why suitable types are important and we fix wrong data types.
+To be more clear about the types, we show the text output for DataFrames instead of a pretty table.
+In this section, we work with the following dataset:
+
+```jl
+@sco process=string post=output_block wrong_types()
+```
+
+Because the date column has the wrong type, sorting won't work correctly:
+
+```jl
+s = "sort(wrong_types(), :date)"
+scsob(s)
+```
+
+To fix the sorting, we can use `Date` as described in @sec:dates:
+
+```jl
+@sco process=string post=output_block fix_date_column(wrong_types())
+```
+
+Now, sorting will work correctly:
+
+```jl
+s = """
+    df = fix_date_column(wrong_types())
+    sort(df, :date)
+    """
+scsob(s)
+```
+
+For the age column, we have a similar problem:
+
+```jl
+s = "sort(wrong_types(), :age)"
+scsob(s)
+```
+
+This isn't right, because an infant is younger than adults and adolescents.
+The solution for this categorical data problem is to use `CategoricalArrays.jl`:
```

Comment:
yet, another package. I feel sometimes that Julia is like a zoo, one needs to know what species is needed. This is just a general remark, people coming from other languages are not use to this. They kinda of expect to have everything loaded and working. Probably we could say something about this somewhere?

Reply:
Compared to R or Python, it's not so much worse, I think. There, they even create a package which groups all the packages :rofl: (tidyverse). 

Compared to C++ or C, it indeed is worse, but in C++ you would just implement everything yourself because managing dependencies is a huge struggle.

> Probably we could say something about this somewhere?

You can open an issue if you want :smile: